instrumentation/openfeign-9.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/openfeign/OpenFeignAttributesGetter.java

```diff
@@ -0,0 +1,122 @@
+import io.opentelemetry.instrumentation.api.instrumenter.http.HttpClientAttributesGetter;
+import io.opentelemetry.instrumentation.api.instrumenter.net.NetClientAttributesGetter;
+import io.opentelemetry.instrumentation.api.instrumenter.rpc.RpcAttributesGetter;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.Nullable;
+
+enum OpenFeignAttributesGetter
+    implements
+        RpcAttributesGetter<ExecuteAndDecodeRequest>,
+        NetClientAttributesGetter<ExecuteAndDecodeRequest, feign.Response>,
+        HttpClientAttributesGetter<ExecuteAndDecodeRequest, feign.Response> {
+  INSTANCE;
+
+  @Override
+  public String system(ExecuteAndDecodeRequest request) {
+    return "openfeign";
+  }
+
+  @Override
+  public String service(ExecuteAndDecodeRequest request) {
+    return request.getTarget().name();
+  }
+
+  @Override
+  public String method(ExecuteAndDecodeRequest request) {
+    return request.getRequestTemplate().method();
+  }
+
+  @Override
+  public List<String> requestHeader(ExecuteAndDecodeRequest request, String name) {
+    Request req = request.getRequestTemplate().request();
+    return new ArrayList<>(req.headers().getOrDefault(name, Collections.emptyList()));
+  }
+
+  @Nullable
+  @Override
+  public Long requestContentLength(ExecuteAndDecodeRequest request, @Nullable Response response) {
+    byte[] body = request.getRequestTemplate().request().body();
+    return body != null ? (long) body.length : null;
+  }
+
+  @Nullable
+  @Override
+  public Long requestContentLengthUncompressed(
+      ExecuteAndDecodeRequest request, @Nullable Response response) {
+    return null;
+  }
+
+  @Override
+  public Integer statusCode(ExecuteAndDecodeRequest request, @Nullable Response response) {
+    return response != null ? response.status() : 0;
+  }
+
+  @Override
+  public Long responseContentLength(ExecuteAndDecodeRequest request, @Nullable Response response) {
+    return (response != null && response.body() != null) ? response.body().length().longValue() : 0;
+  }
+
+  @Nullable
+  @Override
+  public Long responseContentLengthUncompressed(
+      ExecuteAndDecodeRequest request, @Nullable Response response) {
+    return null;
+  }
+
+  @Override
+  public List<String> responseHeader(
+      ExecuteAndDecodeRequest request, @Nullable Response response, String name) {
+    return response != null
+        ? new ArrayList<>(response.headers().getOrDefault(name, Collections.emptyList()))
+        : Collections.emptyList();
+  }
+
+  @Override
+  public String url(ExecuteAndDecodeRequest request) {
+    return request.getTarget().url() + request.getTemplateUri().getPath();
+  }
+
+  @Nullable
+  @Override
+  public String flavor(ExecuteAndDecodeRequest request, @Nullable Response response) {
+    return null;
+  }
+
+  @Override
+  public String transport(ExecuteAndDecodeRequest request, @Nullable Response response) {
+    return "http";
```

Comment:
`http` isn't a valid value for `net.transport`, see https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/span-general.md#general-network-connection-attributes for valid values. Probably you should use `SemanticAttributes.NetTransportValues.IP_TCP`.